lib/desktop/quicksettings/quick_settings.dart

```diff
@@ -341,83 +342,46 @@ class QuickSettingsState extends State<QuickSettings> {
-                      Pangolin.settingsBox.put("language", "nl_NL");
-                      break;
-                    case "nl_NL":
-                      Pangolin.setLocale(context, Locale("fr", "FR"));
-                      Pangolin.settingsBox.put("language", "fr_FR");
-                      break;
-                    case "fr_FR":
-                      Pangolin.setLocale(context, Locale("de", "DE"));
-                      Pangolin.settingsBox.put("language", "de_DE");
-                      break;
-                    case "de_DE":
-                      Pangolin.setLocale(context, Locale("id", "ID"));
-                      Pangolin.settingsBox.put("language", "id_ID");
-                      break;
-                    case "id_ID":
-                      Pangolin.setLocale(context, Locale("pl", "PL"));
-                      Pangolin.settingsBox.put("language", "pl_PL");
-                      break;
-                    case "pl_PL":
-                      Pangolin.setLocale(context, Locale("pt", "BR"));
-                      Pangolin.settingsBox.put("language", "pt_BR");
-                      break;
-                    case "pt_BR":
-                      Pangolin.setLocale(context, Locale("ru", "RU"));
-                      Pangolin.settingsBox.put("language", "ru_RU");
-                      break;
-                    case "ru_RU":
-                      Pangolin.setLocale(context, Locale("sv", "SE"));
-                      Pangolin.settingsBox.put("language", "sv_SE");
-                      break;
-                    case "sv_SE":
-                      Pangolin.setLocale(context, Locale("uk", "UA"));
-                      Pangolin.settingsBox.put("language", "uk_UA");
-                      break;
-                    case "uk_UA":
-                    default:
-                      Pangolin.setLocale(context, Locale("en", "US"));
-                      Pangolin.settingsBox.put("language", "en_US");
-                      break;
+                buildTile(Icons.airplanemode_inactive,
+                    LocaleStrings.pangolin.qsAirplanemode, changeColor),
+                buildTile(Icons.invert_colors_off,
+                    LocaleStrings.pangolin.qsInvertcolors, changeColor),
+                buildTile(
+                    Icons.language, LocaleStrings.pangolin.qsChangelanguage,
+                    () {
+                  int newLocale = 0;
+                  // final currentLanguage = context.locale.countryCode;
+                  for (var locale in Locales.supported) {
+                    if (locale == context.locale) {
+                      final languageIndex =
+                          (newLocale + 1 < Locales.supported.length)
+                              ? newLocale + 1
+                              : 0;
+                      context.locale = Locales.supported[languageIndex];
+                      Pangolin.settingsBox.put("language", "${context.locale}");
```

Comment:
also this should not be done, easy_localization already stores your changes using shared prefs

Reply:
got it. removed languages hive store

Reply:
I did remove it from other calls for it, in genmanagement.dart too.